### DISCUSSION TITLE
Registration of custom callbacks in LightningCLI (PL 1.9.0)

### DISCUSSION THREAD

**Discussion by machur** [Lightning-AI/pytorch-lightning#17602] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hi, I'm using pytorch-lightning in version 1.9.0 and I'm struggling to add my own custom callback to LightningCLI. I've seen that in earlier versions there were a couple of different registries to do so (including CALLBACK_REGISTRY). It looks like that mechanism has been removed in PL version 1.9.0 and higher. Would you please advice?

**Comment by mauvilsa** [comment]
It is not clear what exactly your are struggling with. Your custom callback you can always provide by giving the full import path. If you want to be able to specify only by name, just import the module. This is explained at https://lightning.ai/docs/pytorch/stable/cli/lightning_cli_intermediate_2.html#classes-from-any-package

**Reply by machur** [reply]
@mauvilsa it turned out my code has been working just fine. I was running my script with "--print_config" flag and I was checking printed config instead of the config.xml file saved in lightning_logs folder.  Also with "--print_config" the "before_XXX" hooks were not called, so it resulted in more confusion on my side.

Thanks for your help.
